core/src/main/java/com/linecorp/armeria/client/Http2ResponseDecoder.java

```diff
@@ -96,9 +96,9 @@ private void onWrapperCompleted(HttpResponseWrapper resWrapper, int id, @Nullabl
         resWrapper.onSubscriptionCancelled(cause);
 
         if (cause != null) {
-            // We are not closing the connection but just send a RST_STREAM,
-            // so we have to remove the response manually.
-            removeResponse(id);
+            // Removing the response and decrementing {@code unfinishedResponses} isn't done immediately
+            // here. Instead, we rely on {@code Http2ResponseDecoder#onStreamClosed} to decrement
+            // `unfinishedResponses` to match the timing where netty decrements {@code numActiveStreams}.
```

Comment:
How about?
```suggestion
            // `unfinishedResponses` after Netty decrements `numActiveStreams` in `DefaultHttp2Connection` so that `unfinishedResponses` is never greater than `numActiveStreams`.
```